NetCord/User.cs

```diff
@@ -2,43 +2,230 @@
+    /// </summary>
+    /// <remarks>
+    /// Requires the <c>identify</c> OAuth2 scope.
+    /// </remarks>
     public string? AvatarDecorationHash => _jsonModel.AvatarDecorationHash;
 
+    /// <summary>
+    /// Whether the user has a set custom avatar.
+    /// </summary>
     public bool HasAvatar => AvatarHash is not null;
 
-    public ImageUrl GetAvatarUrl(ImageFormat? format = null) => ImageUrl.UserAvatar(Id, AvatarHash!, format);
+    /// <summary>
+    /// Gets the <see cref="ImageUrl"/> of the user's avatar.
+    /// </summary>
+    /// <param name="format">The format of the returned <see cref="ImageUrl"/>. Defaults to <see cref="ImageFormat.Png"/> (or <see cref="ImageFormat.Gif"/> for animated avatars).</param>
+    /// <returns>An <see cref="ImageUrl"/> pointing to the user's avatar. If the user does not have one set, returns <see langword="null"/>.</returns>
+    public ImageUrl? GetAvatarUrl(ImageFormat? format = null) => AvatarHash is string hash ? ImageUrl.UserAvatar(Id, hash, format) : null;
 
+    /// <summary>
+    /// Whether the user has a set custom banner image.
+    /// </summary>
     public bool HasBanner => BannerHash is not null;
 
-    public ImageUrl GetBannerUrl(ImageFormat? format = null) => ImageUrl.UserBanner(Id, BannerHash!, format);
+    /// <summary>
+    /// Gets the <see cref="ImageUrl"/> of the user's banner.
+    /// </summary>
+    /// <param name="format">The format of the returned <see cref="ImageUrl"/>. Defaults to <see cref="ImageFormat.Png"/> (or <see cref="ImageFormat.Gif"/> for animated banners).</param>
+    /// <returns>An <see cref="ImageUrl"/> pointing to the user's banner. If the user does not have one set, returns <see langword="null"/>.</returns>
+    public ImageUrl? GetBannerUrl(ImageFormat? format = null) => BannerHash is string hash ? ImageUrl.UserBanner(Id, hash, format) : null;
 
+    /// <summary>
+    /// Whether the user has a set avatar decoration.
+    /// </summary>
     public bool HasAvatarDecoration => AvatarDecorationHash is not null;
 
-    public ImageUrl GetAvatarDecorationUrl() => ImageUrl.UserAvatarDecoration(Id, AvatarDecorationHash!);
+    /// <summary>
+    /// Gets the <see cref="ImageUrl"/> of the user's avatar decoration URL.
+    /// </summary>
+    /// <returns>An <see cref="ImageUrl"/> pointing to the user's avatar decoration. If the user does not have one set, returns <see langword="null"/>.</returns>
+    public ImageUrl? GetAvatarDecorationUrl() => AvatarDecorationHash is string hash ? ImageUrl.UserAvatarDecoration(Id, hash) : null;
 
+    /// <summary>
+    /// Returns an <see cref="ImageUrl"/> object representing the user's default avatar.
+    /// </summary>
     public ImageUrl DefaultAvatarUrl => Discriminator is 0 ? ImageUrl.DefaultUserAvatar(Id) : ImageUrl.DefaultUserAvatar(Discriminator);
 
+    /// <summary>
+    /// Converts the ID of this user into its string representation, using Discord's mention syntax.
```

Comment:
Too confusing I think

Reply:
This is how .NET in general describes it's `.ToString()` methods, but I added a small clarifcation.